pkg/primaza/workercluster/agentsvc.go

```diff
@@ -140,6 +133,22 @@ spec:
           requests:
             cpu: 10m
             memory: 64Mi
-      serviceAccountName: primaza-agentsvc
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+        volumeMounts:
+        - mountPath: /tmp/k8s-webhook-server/serving-certs
+          name: cert
+          readOnly: true
+      securityContext:
+        runAsNonRoot: true
+      serviceAccountName: primaza-controller-agentsvc
```

Comment:
```suggestion
      serviceAccountName: primaza-agentsvc
```

Reply:
Q: the application agent deploys under `primaza-controller-agentapp`, why make this change?

Reply:
some inconsistencies in tests. This is the less invasive fix. Once we will introduce primazactl we will fix this naming issues

Reply:
nvm, this is the service account, not the deployment.

Reply:
Fixed, though this may need more changes in the future.